src/sagemaker/serve/validations/optimization.py

```diff
@@ -0,0 +1,165 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Holds the validation logic used for the .optimize() function"""
+from typing import Any, Dict, Set
+from enum import Enum
+from pydantic import BaseModel
+import textwrap
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class OptimizationContainer(Enum):
```

Comment:
We can use these for internal organization. But we should not burden customers with knowing how our optimizations work, which containers are used under which scenarios, etc.

Consider the following error messages - 
- Optimizations that use Compilation and Speculative Decoding are not currently supported on GPU & Neuron instances.
- Optimizations that use Compilation and Quantization are not currently supported on Neuron instances.
- Optimizations that use Quantization:SmoothQuant are only supported with Compilation on GPU instances.


The general theme we want to maintain is 

Optimizations {A,B,C} are {not|only} supported with optimizations {X,Y,Z} on instances {L,M,N}